snowfakery/parse_recipe_yaml.py

```diff
@@ -315,16 +325,46 @@ def parse_object_template(yaml_sobj: Dict, context: ParseContext) -> ObjectTempl
         return new_template
 
 
-def parse_object_template_list(
-    object_templates: List[dict], context: ParseContext
-) -> List[ObjectTemplate]:
-    parsed_object_templates = []
-    for obj in object_templates:
+def parse_variable_definition(
+    yaml_sobj: Dict, context: ParseContext
+) -> VariableDefinition:
+    parsed_template: Any = parse_element(
+        yaml_sobj,
+        "var",
+        {},
+        {
+            "value": (str, int, dict, list),
+        },
+        context,
+    )
+
+    assert yaml_sobj
+    with context.change_current_parent_object(yaml_sobj):
+        sobj_def = {}
+        sobj_def["varname"] = parsed_template.var
```

Comment:
Hmm, it's typed as `Any`, but we obviously expect it to have a `var`. Doesn't need to be in this PR, but perhaps we can replace `parse_element` with something that is typed more specifically.

Reply:
parse_element is the boundary between the untyped YAML world and the dynamically typed world. It's more or less my "poor man's Pydantic". So its return type will always be polymorphic.

Reply:
@prescod ok, I get that parse_element needs to return different types. For a case like this where a particular type is _expected_, it would still be possible to add a type hint on the variable that the return value is being assigned to (along with perhaps an assertion to confirm that it's the expected type at runtime).

Anyway, not clear that's worth the effort here. I'll resolve this.